tests/test_search_adaptors.py

```diff
@@ -510,3 +511,33 @@ def test_vespa_search_adaptor__metadata(test_vespa, query_string, metadata_filte
             assert hit.metadata not in [None, []]
             for metadata_filter in metadata_filters:
                 assert metadata_filter in hit.metadata
+
+
+@pytest.mark.vespa
+@pytest.mark.parametrize(
+    "query_string, filters",
+    [
+        (
+            "the",
+            {"family_geographies": ["BIH"]},
+        ),
+        (
+            "the",
+            {"family_geographies": ["BIH", "NOR"]},
+        ),
+    ],
+)
+def test_vespa_search_adaptor__filters(test_vespa, query_string, filters):
+    """Test that the search filters work"""
+    request = SearchParameters(
+        query_string=query_string,
+        filters=Filters.model_validate(filters),
+    )
+    response = vespa_search(test_vespa, request)
+    assert response.total_family_hits > 0
+    for family in response.families:
+        for hit in family.hits:
+            for filter_name, filter_values in filters.items():
+                attribute_value_from_hit = getattr(hit, filter_name)
+                assert attribute_value_from_hit not in [None, []]
+                assert all([val in attribute_value_from_hit for val in filter_values])
```

Comment:
nitpick: I'm finding the nested for loop difficult to parse. Especially as there are two objects being iterated (response & filters). I think there are ways to make this more readable that would be worth a moment exploring for when we revisit. One/both/other of:
- creating hits in advance with a list comprehension
- Use a dedicated function to get the attribute from the hit instead of relying on its scope within the loop

Reply:
> ...also add negative condition somewhere in here making sure we aren't getting families without those geographies in them?

`assert all([val in attribute_value_from_hit for val in filter_values])`

So this line here checks that ever filter value exists in the `hit.family_geographies`. 

E.g. We filtered for `["BIH", "NOR"]`, do `"BIH"` and `"NOR"` exist in `hit.family_geographies`

So from my understanding we would throw an `AssertionError` if a family didn't have those geographies.

Reply:
Ah nice, yep that checks out 👍 thanks!